enhancements/oc/day2-add-nodes.md

```diff
@@ -0,0 +1,271 @@
+hardware, and by applying a number of live pre-flight validations in the target environment to spot in advance those issues that could hamper 
+the setup. Also, capturing all the necessary elements for the installation in a single ISO image simplified the application and distribution 
+into different environments. 
+
+The current enhancement proposes a standard and unified way for adding nodes to an on-premises cluster, by replicating the same approach 
+successfully adopted for the installation process, so that the end user could benefit from a more consistent and simple workflow. In addition, 
+the standardization will make obsolete the hand-rolled UPI method, the need to use MCE to manage local clusters, and (unsupported) field-developed 
+automations that are expensive to maintain.
+
+### User Stories
+- As a cluster administrator, I want to add (or replace) one or more nodes to an existing cluster (potentially deployed in a disconnected environment) 
+  by generating an ISO and use it to boot the required nodes
+- As a cluster administrator, I want to provide the initial static network configuration required for each different host that will be added to the cluster
+- As a cluster administrator, I want to validate the nodes to be added before initiating the procedure and be informed of any validation failures
+- As a cluster administrator, I want to be able to monitor the process of adding new nodes
+
+### Goals
+- Allow the user to easily add one or more worker nodes to an existing (disconnected) cluster where MachineSet scaleout is not available or feasible, 
+  with the minimum amount of configuration requested
+- Provide a simpler interface by reusing existing command line tools for cluster administration
+- When adding multiple nodes, a failure in adding a node should not impact the other nodes currently involved in the process
+- Cover at least Baremetal, vSphere, Nutanix and None platform (regardless of the installation method used)
+- For IPI clusters, provide support for the scenario where the user can’t use the hosts BMCs
+
+### Non-Goals
+- Replace MachineSet-based management of hosts for IPI clusters
+- Offer more or finer-grained configuration options for new nodes than those supported by the agent-based installer
+- Generation of the ISO image for adding the nodes into an environment not directly connected to the target cluster
+- Provide an automation for final nodes approval
+
+## Proposal
+A key factor for simplifying the user experience consists in offering a simple enough - but sufficiently flexible -  user interface 
+to accomplish the task of adding the required nodes to a cluster, and to monitor the progress of such activity. For this reason, 
+the proposal consists into enhancing the `oc` cli with two new commands:
+
+- `oc adm add-nodes-image` to generate the ISO for adding the nodes to the cluster
+- `oc adm monitor-add-nodes` to monitor the nodes currently being added to the cluster
```

Comment:
I'd consider introducing a single command with either separate flags or sub-commands handling adding vs monitoring. Maybe something like `oc adm add-node-images` which would either check for an existing namespace and report progress or create a new invocation. I'm suspecting that don't want to allow running more than a single instance at any point in time. Additionally some kind of `--wait` flag which would allow the user to just stream/watch the logs as it goes.

Reply:
That's definitely worth considering, but the reason for splitting it up is that there is a manual step (booting the ISO you got in the first part) in between the two parts.
Furthermore, there is data required as input to the second part that may not be available when the first part is invoked - if you are using DHCP then the easiest time to get the IP address is _after_ you've booted the server with the ISO.

I don't think running multiple instances would hurt, the more important thing is that we clean them up when they're done.